test_case_gen.py

```diff
@@ -0,0 +1,249 @@
+              link_buffer=LinkBuffer(max_buffer_size_bits=64.0 * KILOBYTE),
+              static_delay_sec=0.01, capacity_bps=10.0 * MEGABIT)
+    l5 = Link(name="L5", end_1_addr="R4", end_2_addr="H2",
+              link_buffer=LinkBuffer(max_buffer_size_bits=64.0 * KILOBYTE),
+              static_delay_sec=0.01, capacity_bps=12.5 * MEGABIT)
+    links.append(l0)
+    links.append(l1)
+    links.append(l2)
+    links.append(l3)
+    links.append(l4)
+    links.append(l5)
+
+    # Dummy
+    flows = list()
+    f1 = FlowDummy(flow_id="F1", source_addr="H1", dest_addr="H2",
+                   data_size_bits=20.0 * MEGABYTE,
+                   start_time_sec=0.5)
+    flows.append(f1)
+    network = NetworkTopology(links=links, flows=flows, hosts=hosts,
+                              routers=routers)
+    network.write_to_json("data/test_case_1_dummy.json")
+
+    # Reno
+    flows = list()
+    f1 = FlowReno(flow_id="F1", source_addr="H1", dest_addr="H2",
+                  data_size_bits=20.0 * MEGABYTE,
+                  start_time_sec=0.5)
+    flows.append(f1)
+    network = NetworkTopology(links=links, flows=flows, hosts=hosts,
+                              routers=routers)
+    network.write_to_json("data/test_case_1_reno.json")
+
+    # FAST
+    flows = list()
+    f1 = FlowFast(flow_id="F1", source_addr="H1", dest_addr="H2",
+                  data_size_bits=20.0 * MEGABYTE,
+                  start_time_sec=0.5)
+    flows.append(f1)
+    network = NetworkTopology(links=links, flows=flows, hosts=hosts,
+                              routers=routers)
+    network.write_to_json("data/test_case_1_fast.json")
+
+    # test case 2
+    hosts = list()
+    s1 = Host("S1")
+    s2 = Host("S2")
+    s3 = Host("S3")
+    t1 = Host("T1")
+    t2 = Host("T2")
+    t3 = Host("T3")
+    hosts.append(s1)
+    hosts.append(s2)
+    hosts.append(s3)
+    hosts.append(t1)
+    hosts.append(t2)
+    hosts.append(t3)
+
+    routers = list()
+    r1 = Router("R1")
+    r2 = Router("R2")
+    r3 = Router("R3")
+    r4 = Router("R4")
+    routers.append(r1)
+    routers.append(r2)
+    routers.append(r3)
+    routers.append(r4)
+
+    # router-router
+    links = list()
+    ls1 = Link(name="LS1", end_1_addr="R1", end_2_addr="R2",
+              link_buffer=LinkBuffer(max_buffer_size_bits=128.0 * KILOBYTE),
+              static_delay_sec=0.01, capacity_bps=10.0 * MEGABIT)
+    ls2 = Link(name="LS2", end_1_addr="R2", end_2_addr="R3",
+              link_buffer=LinkBuffer(max_buffer_size_bits=128.0 * KILOBYTE),
+              static_delay_sec=0.01, capacity_bps=10.0 * MEGABIT)
+    ls3 = Link(name="LS3", end_1_addr="R3", end_2_addr="R4",
+              link_buffer=LinkBuffer(max_buffer_size_bits=128.0 * KILOBYTE),
+              static_delay_sec=0.01, capacity_bps=10.0 * MEGABIT)
+    links.append(ls1)
+    links.append(ls2)
+    links.append(ls3)
+    lh1 = Link(name="LH1", end_1_addr="S1", end_2_addr="R1",

```

Comment:
Name the router-host link that connects to "S1" as "LS1". Similarly, the router-host link that connects to "T1" should be called "LT1", etc. Same for S2, S3, T2, and T3.